coderd/workspaces.go

```diff
@@ -1190,8 +1190,11 @@ func (api *API) putWorkspaceTTL(rw http.ResponseWriter, r *http.Request) {
 
 			if build.Transition == database.WorkspaceTransitionStart {
 				if err = s.UpdateWorkspaceBuildDeadlineByID(ctx, database.UpdateWorkspaceBuildDeadlineByIDParams{
-					ID:          build.ID,
-					Deadline:    time.Time{},
+					ID: build.ID,
+					// If the build has a max_deadline set, we still need to
+					// abide by it. It will either be zero (our target), or a
+					// value.
+					Deadline:    build.MaxDeadline,
```

Comment:
Not sure I follow the reasoning behind this change 🤔 
According to the above comment:
```
// If autostop has been disabled, we want to remove the deadline from the
// existing workspace build (if there is one).
if !dbTTL.Valid {
```
So, shouldn't deadline be `time.Time{},`?

Reply:
Additionally, if I understood correctly, `Deadline` is derived from TTL and `MaxDeadline` is derived from Autostop. In that case, when autostop is disabled, shouldn’t both `Deadline` and `MaxDeadline` be unset? 🤔

Reply:
A new build with `ttl: 0, autostop_requirement: enabled` will end up with the deadlines of `deadline: required_autostop_time, max_deadline: required_autostop_time` (e.g. equal deadline and max_deadline).

So this matches that behavior.